app/src/main/java/activity/DetailsActivity.java

```diff
@@ -0,0 +1,163 @@
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ListView;
+import android.widget.TextView;
+import android.widget.Toast;
+import com.fln.flnpublictransportation.R;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import java.util.ArrayList;
+import java.util.List;
+import controlers.RestServices;
+import controlers.StopAdapter;
+import dto.Departure;
+import dto.Stop;
+
+public class DetailsActivity extends Activity {
+    private String id;
+    private String searchKey;
+    private StopAdapter stopAdapter;
+    private String strWeekDay = "";
+    private String strSaturday = "";
+    private String strSunday = "";
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_details);
+
+        Bundle params = getIntent().getExtras();
+        id = params.getString("id");
+        searchKey = params.getString("searchKey");
+        TextView txtTitle = (TextView) findViewById(R.id.details_title_route);
+        txtTitle.setText(params.getString("name"));
+
+        Button btnBack = (Button) findViewById(R.id.button_back);
+        btnBack.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(DetailsActivity.this, MainActivity.class);
+                Bundle params = new Bundle();
+                params.putString("searchKey", searchKey);
+                intent.putExtras(params);
+                startActivity(intent);
+                finish();
+            }
+        });
+
+        stopAdapter  = new StopAdapter(new ArrayList<Stop>(), this);
+        ListView listViewStops = (ListView) findViewById(R.id.list_stops);
+        listViewStops.setAdapter(stopAdapter);
+
+        (new AsyncListStopsLoader()).execute(getString(R.string.findStopsByRouteId));
+        (new AsyncListDeparturesLoader()).execute(getString(R.string.findDeparturesByRouteId));
+    }
+
+    private class AsyncListStopsLoader extends AsyncTask<String, Void, List<Stop>> {
+
+        @Override
+        protected void onPostExecute(List<Stop> listStop) {
+            super.onPostExecute(listStop);
+            stopAdapter.setItemList(listStop);
+            stopAdapter.notifyDataSetChanged();
+            if(listStop == null || listStop.isEmpty()) {
+                Toast.makeText(getApplicationContext(), R.string.no_stops, Toast.LENGTH_LONG).show();
+            }
+        }
+
+        @Override
+        protected void onPreExecute() {
+            super.onPreExecute();
+        }
+
+        @Override
+        protected List<Stop> doInBackground(String... params) {
+            List<Stop> listStop = new ArrayList<Stop>();
+
+            try {
+                RestServices restService = new RestServices();
+                String param =  "{\"params\":{\"routeId\": \"" + id + "\"}}";
```

Comment:
You have both param and params variable in the same scope. It's a bit confusing. Try to use more specific names.